### DISCUSSION TITLE
The furigana on top of お願い致します is too verbose

### DISCUSSION THREAD

**Discussion by Morgawr** [Jitendex/Jitendex#40] [Errors and Corrections]
Looking at this expression in yomitan: 

![image](https://github.com/stephenmk/Jitendex/assets/184056/b6d67a1c-18af-4d23-bb82-cc36d72b887b)

Normally the furigana for お願い致します would only have ねが on top of 願 and いた on top of 致, but in this particular entry it seems like the entire furigana is displayed, including those of kana that is already present.

**Comment by stephenmk** [comment]
The furigana in that text [is produced by the yomitan extension](https://github.com/themoeway/yomitan/blob/55897b2b29e88ffd0c9140d03b9e74c4a94d98bd/ext/js/language/sandbox/japanese-util.js) and there's nothing I can do with the dictionary data to change it. They'd need to rework their furigana distribution algorithm somehow.

The problem here is that there is more than one way you can distribute those kana over that expression. 
"おねがいいたします" over "お願い致します" could either be distributed as "お<ruby>願<rt>ねが</rt></ruby>い<ruby>致<rt>いた</rt></ruby>します" (correct) or "お<ruby>願<rt>ねがい</rt></ruby>い<ruby>致<rt>た</rt></ruby>します" (incorrect). Since yomitan has no way of knowing which is correct, it just places all of the kana over the whole expression.

Within the dictionary data itself, I [implemented my own program](https://github.com/stephenmk/jitenbot/blob/main/tests/test_furigana.py) for distributing furigana over expressions. So for example, Yomitan can't correctly apply the furigana over the expression 好き嫌い (it could either be <ruby>好<rt>すき</rt></ruby>き<ruby>嫌<rt>ら</rt></ruby>い or <ruby>好<rt>す</rt></ruby>き<ruby>嫌<rt>きら</rt></ruby>い) but within the dictionary data it is done correctly.

<details>
<summary>好き嫌い in a cross reference</summary>

![1](https://github.com/stephenmk/Jitendex/assets/8003332/84966ecb-ca09-440f-a800-bddd9893789d)
</details>

<details>
<summary>好き嫌い displayed correctly in an example sentence but not in the Yomitan headword</summary>

![2](https://github.com/stephenmk/Jitendex/assets/8003332/4838dcc8-600d-452d-a3be-05e7f55cd1f1)
</details>

I'm using data from [the JMdictFurigana project](https://github.com/Doublevil/JmdictFurigana) to assist my program with these ambiguous cases. The data from that project eats up a few dozen megabytes of space, so it might not be easily integrated with Yomitan's parser.

**Comment by Morgawr** [comment]
Oh, I see where the problem is, the problem is because of the repeated kana, right? So it only happens in expressions that are like おねが**いい**たします or す**きき**らい. For example I looked at other mixed kana/kanji expressions like 我が儘 vs 我儘 vs 我がまま and it doesn't happen. 

Hmm this is tricky. I'll try file an issue with the yomitan folks and see if someone wants to pick it up. Thanks for explaining where the problem was.

**Reply by stephenmk** [reply]
> the problem is because of the repeated kana, right?

The kana don't necessarily have to be repeated immediately, and the problem doesn't occur just because kana are repeated. The problem is because the kana can be distributed over the kanji in multiple ways that appear to be equally valid to Yomitan.

命の水 (いのちのみず) doesn't have sequentially repeated kana, but the there are two different ways you can distribute the kana. It could either be

### <ruby>命<rt>いのち</rt></ruby>の<ruby>水<rt>みず</rt></ruby></span>
or
### <ruby>命<rt>い</rt></ruby>の<ruby>水<rt>ちのみず</rt></ruby>
